library/include/borealis/core/touch/pan_gesture.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+    Copyright 2021 XITRIX
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#include <borealis/core/gesture.hpp>
+
+namespace brls
+{
+
+class PanGestureRecognizer;
+typedef std::function<void(PanGestureRecognizer*)> PanGestureRespond;
+
+// Axis of pan recognition start conditions
+enum class PanAxis
+{
+    HORIZONTAL, // Triggers only on horizontal coordinate changes
+    VERTICAL, // Triggers only on vertical coordinate changes
+    NONE, // Any movement allowed
+};
+
+struct position
```

Comment:
What do you think about implementing structures for view's frame? 
i.e. instead of 4 vars (x, y, width, height) create something like:

struct Position { x, y }
struct Size { width, height }
struct Rect { Position, Size }

and use Rect to store all those vars.

I think they could be useful, also I could use one here.

Reply:
The thing is that a view does not have a width or height stored, it's stored internally in the yoga node and returned by `YGNodeLayoutGetWidth` when we need it.

Unless you changed that and I haven't got to that point in reading the PR yet?

Sorry for the long delay I was in vacation 🌴 

I will resume the review in a few days when I wrap things up

Reply:
Hope you had a good rest!

No I haven't changed it yet. I've made another brunch where I tried to implement that idea and I found that it's yoga's node variables :\ 

But still I think we could change lot's of: 
float originX; 
float originY; 

to one:
Point origin;

I think those structs will be useful for lots of another things. When I'll finish with that idea, I'll create a PR for it.

Reply:
I mean yeah, I want to avoid boilerplate though, how many times and where do you think we could use that new struct in the code?

Reply:
I'm not sure about Size and Rect, but Point could be used everywhere: View's detachedOrigin, translation, Touch\Mouse position, velocity, and lots of other stuff, which contains X and Y variables.

Reply:
Go for it then I guess

Reply:
Should I do that in separate brunch, or I can do it here?

Reply:
You can do it here 👍